### PR TITLE
[TTS] MagpieTTS: Implement Frechet Codec Distance metric + some minor inference bugfixes

### PR DESCRIPTION
**Pull request by rfejgin**
# What does this PR do ?

Adds the Frechet Codec Distance metric and integrates it in MagpieTTS inference scripts. Also fixes some minor MagpieTTS inference bugs.

**Collection**: TTS

# Changelog

The Frechet Distance (FD) is commonly used to evaluate generative models (e.g. Frechet Inception Distance, Frechet Audio Distance). In this PR we implements  FD in the embedding space of a neural codec. This is a metric that measures how closely the distributions of real and generated codec frames match, at the single frame level.

Changes:
* `frechet_codec_distance.py`: An implementation of FD in codec embedding space. Builds on TorchMetrics' FID implementation. We provide the audio codec as a custom feature extractor.
* `test_frechet_coec_distance.py`: Unit test
* Integration of the FCD in MagpieTTS inference scripts. If desired, FCD calculation can be disabled using the `--disable_fcd` command line argument to `magpietts_inference.py`
* Inference bugfixes
  * fix a logging statement that was reporting errors due to incorrect formatting syntax
  * disable logging of thousands of messages during loading of the `titanet_small` speaker representation model. This was present in earlier versions of the inference scripts and appears to have been accidentally lost in recent refactorings 
  * Fix an issue where filewise metrics were not being filtered to a spcified subset as intended

**PR Type**:

- [x] New Feature
- [x] Bugfix
- [ ] Documentation

### CODE REVIEW

File: nemo/collections/tts/metrics/frechet_codec_distance.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        # We treat all frames as one large batch element, since the codec requires (B, C, T) input and
+        # we don't have the per-batch-element lengths at this point due to FID API limitations
+
+        # Consturct a length tensor: one batch element, all frames.
+        x_len = torch.tensor(x.shape[0], device=x.device, dtype=torch.long).unsqueeze(0)  # (1, 1)
+        tokens = x.permute(1, 0).unsqueeze(0)  # 1, C, B*T
+        embeddings = self.codec.dequantize(tokens=tokens, tokens_len=x_len)  # (B, D, T)
+        # we treat each time step as a separate example
+        embeddings = rearrange(embeddings, 'B D T -> (B T) D')
+        return embeddings
```
Comment (blisc) [inline nemo/collections/tts/metrics/frechet_codec_distance.py:59]: Don't we need some sort of masking here? If we are reducing B and T into one dimension, how do we ensure that no padding gets passed to the model?

Comment (blisc) [reply nemo/collections/tts/metrics/frechet_codec_distance.py:59]: If this only works on batch 1, we should add a check that x has a batch dimension of 1 or output an error or a warning

**Review by blisc — CHANGES_REQUESTED** [review]

File: nemo/collections/tts/metrics/frechet_codec_distance.py (hunk 1 above)
Comment (rfejgin) [reply nemo/collections/tts/metrics/frechet_codec_distance.py:59]: It works for all batch sizes. 

Removal of padding is taken care of in [FrechetCodecDistance.update()](https://github.com/NVIDIA-NeMo/NeMo/pull/15223/changes#diff-7d4075bbfa3672447ab46a1aa0d746f9f4a70b2429175e255341e53db9848128R146)

The way it works is:
`fcd.update()` constructs a large batch of shape `(B*T, C)` which eventually gets routed to the `CodecEmbedder`. All frames in this batch are valid; padding is explicitly removed. The `CodecEmbedder` then internally reshapes the tensor to `1,C, B*T` before feeding into the codec for dequantization.

The reason we need to do it this way is that `FrechetInceptionDistance`'s APIs don't allow passing in a second tensor indicating which frames are valid and not (as we would usually do using `tokens_len`); so to work around that we pass in a flattened tensor all of whose entries are valid.

**Comment by rfejgin** [comment]
@subhankar-ghosh Could you please review just the latest commit in this PR? That part touches EvaluationConfig and longform inference. I mean the commit titled `Integrate FCD in longform inference and rework --disable_fcd`. Thanks!

File: nemo/collections/tts/modules/magpietts_inference/evaluate_generated_audio.py [hunk 2]
```diff
@@ -240,10 +243,9 @@ def evaluate(
     speaker_verification_model_alternate = speaker_verification_model_alternate.to(device)
     speaker_verification_model_alternate.eval()
 
-    if codec_model_path is not None:
+    if with_fcd:
         fcd_metric = FrechetCodecDistance(codec_name=codec_model_path).to(device)
```
Comment (subhankar-ghosh) [inline nemo/collections/tts/modules/magpietts_inference/evaluate_generated_audio.py:247]: I am not sure if `from torchmetrics.image.fid import FrechetInceptionDistance` is installed in NeMo container by default. If it is not then we might want to check for it's availability and based on that log a warning message if it is not installed.

File: nemo/collections/tts/modules/magpietts_inference/evaluate_generated_audio.py [hunk 3]
```diff
@@ -405,7 +407,7 @@ def evaluate(
         fcd = fcd_metric.compute().cpu().item()
         fcd_metric.reset()
     else:
-        fcd = 0.0
+        fcd = float('nan')
```
Comment (subhankar-ghosh) [inline nemo/collections/tts/modules/magpietts_inference/evaluate_generated_audio.py:410]: Be careful about setting the None value for metrics, check how it is formatted in `def compute_mean_with_confidence_interval`. We should have a uniform default value for the condition `if <any_metric> is None:` case.

**Review by subhankar-ghosh — APPROVED** [review]

File: nemo/collections/tts/modules/magpietts_inference/evaluate_generated_audio.py (hunk 3 above)
Comment (blisc) [reply nemo/collections/tts/modules/magpietts_inference/evaluate_generated_audio.py:410]: @rfejgin, can you add a test wrapper around this in another PR?

Comment (rfejgin) [reply nemo/collections/tts/modules/magpietts_inference/evaluate_generated_audio.py:410]: In [another PR](https://github.com/NVIDIA-NeMo/NeMo/pull/15213) @blisc asked to use `NaN` as a default value for metrics that don't get computed in the specific inference run, so I'm aligning with that. 

Confirmed that `compute_mean_with_confidence_interval` doesn't crash with this NaN value. We end up printing `nan`, which seems reasonable (or if there are repeats `nan +- nan`).

File: nemo/collections/tts/modules/magpietts_inference/evaluate_generated_audio.py (hunk 2 above)
Comment (rfejgin) [reply nemo/collections/tts/modules/magpietts_inference/evaluate_generated_audio.py:247]: Thanks. I double checked and the import proceeds correctly on recent NeMo containers (`nemo:dev` and `nemo:25.11`).

File: nemo/collections/tts/modules/magpietts_inference/evaluate_generated_audio.py (hunk 3 above)
Comment (rfejgin) [reply nemo/collections/tts/modules/magpietts_inference/evaluate_generated_audio.py:410]: > @rfejgin, can you add a test wrapper around this in another PR?
@blisc 

Sure. Do you mean to add a test that checks that when setting `--disable_fcd` the inference script doesn't crash?
